fix: macOS-native binary build broken on main after #4517 service.rs submodule split (30 compile errors)

## Problem

After PR #4517 (`refactor(service): split service.rs into submodules`) merged to `main`, the **macOS-native** `freenet` binary no longer compiles. Pristine `origin/main` (`af18e1f9a`) fails `cargo build -p freenet --bin freenet` on macOS with **30 compile errors**, all in the new `crates/core/src/bin/commands/service/` submodule layout:

```
 6 × E0364  re-export of private item (generate_plist, disable/enable/is_launch_at_login_*,
            macos_app_bundle_path, toggle_launch_at_login_outcome, ...)
 1 × E0365  re-export of private `ToggleLaunchAtLoginOutcome`
11 × E0603  private module / item access
 9 × E0425  cannot find function in scope (first_run_marker_path, is_first_run_at,
            legacy_migration_marker_path, mark_first_run_complete_at, ...)
 2 × E0282  type annotations needed
```

The errors are all **visibility / scope / missing-`use`** fallout from the split: helpers that were free functions in the old monolithic `service.rs` are now in submodules but still referenced unqualified (E0425) or re-exported while `private` (E0364/E0365/E0603).

## Impact

- macOS-native binary does **not** build from `main` — blocks local macOS development and any macOS release/cross build path that compiles natively.
- The Linux build is unaffected (the broken symbols are behind `#[cfg(target_os = "macos")]`), which is almost certainly why #4517's CI passed: **the split's CI did not build the macOS target.**

## Suggested fix

- Bump the visibility of the split-out helpers/types to `pub(super)` / `pub(crate)` so the re-exports in the parent module resolve (fixes E0364/E0365/E0603).
- Add the missing `use crate::commands::service::...` imports (or fully-qualify) at the new call sites (fixes E0425).
- Add the needed type annotations (E0282).
- **Follow-up**: add a macOS build job (or `cargo check --target` for macOS) to CI so a cfg-gated platform path can't merge red again — this is the exact failure mode `.claude/rules/deployment.md` warns about ("a cfg'd path CI does not build on that target is unverified").

Discovered while rebasing #4382 onto the post-#4517 layout; reproduced on pristine `origin/main`.

[AI-assisted]

**Scope correction: the break is not macOS-only — Windows is broken too.**

The same #4517 submodule-split fallout fails the **Windows Check** CI job on unrelated open PRs (#4509, #4522), with the same error classes against the Windows-cfg paths:

```
error[E0364]: `kill_freenet_service_processes` is only public within the crate, and cannot be re-exported
error[E0425]: cannot find function `find_latest_log_file` in this scope   (×3)
error[E0425]: cannot find function `first_run_marker_path` in this scope
error[E0425]: cannot find function `is_first_run_at` in this scope
error[E0425]: cannot find value `FIRST_RUN_OPENER_SPAWNED` in this scope
error[E0282]: type annotations needed
```

So the regression is **all non-Linux platform builds** (macOS native + Windows), not just macOS. Linux CI passed for #4517 because the broken symbols are all behind `#[cfg(target_os = "macos")]` / `#[cfg(target_os = "windows")]`, which the Linux build never compiles.

**Merge-ordering implication:** this fix should land first — it unblocks the Windows Check on #4509 and #4522 (their Windows failures are this regression, not defects in those PRs). The CI follow-up should add **both** a macOS build job and confirm the existing Windows Check guards the `service/` submodule paths.

--- a/crates/core/src/bin/commands/service.rs
+++ b/crates/core/src/bin/commands/service.rs
@@ -187,17 +187,17 @@ fn open_url_in_browser(url: &str) {
 
 /// Path of the first-run marker file, if we can determine a data directory.
 #[allow(dead_code)] // Used by the tray wrapper on Windows/macOS only
-fn first_run_marker_path() -> Option<PathBuf> {
+pub(super) fn first_run_marker_path() -> Option<PathBuf> {
     dirs::data_local_dir().map(|d| d.join("freenet").join(".first-run-complete"))
 }
 
 #[allow(dead_code)]
-fn is_first_run_at(marker: &Path) -> bool {
+pub(super) fn is_first_run_at(marker: &Path) -> bool {
     !marker.exists()
 }
 
 #[allow(dead_code)]
-fn mark_first_run_complete_at(marker: &Path) -> std::io::Result<()> {
+pub(super) fn mark_first_run_complete_at(marker: &Path) -> std::io::Result<()> {
     if let Some(parent) = marker.parent() {
         std::fs::create_dir_all(parent)?;
     }
@@ -211,12 +211,12 @@ fn mark_first_run_complete_at(marker: &Path) -> std::io::Result<()> {
 /// onboarding state. The migration is its own one-shot, tracked by its own
 /// marker, so it runs once for existing DMG users too.
 #[allow(dead_code)]
-fn legacy_migration_marker_path() -> Option<PathBuf> {
+pub(super) fn legacy_migration_marker_path() -> Option<PathBuf> {
     dirs::data_local_dir().map(|d| d.join("freenet").join(".legacy-migration-complete"))
 }
 
 #[cfg(any(target_os = "windows", target_os = "macos"))]
-fn dashboard_port_is_listening() -> bool {
+pub(super) fn dashboard_port_is_listening() -> bool {
     use std::net::TcpStream;
     use std::time::Duration;
     let Ok(addr) = DASHBOARD_ADDR.parse::<std::net::SocketAddr>() else {
@@ -354,14 +354,14 @@ pub use macos::stop_and_remove_service;
 
 // Re-exports for tray.rs (Launch at Login toggle and state query).
 #[cfg(target_os = "macos")]
-pub use launch_at_login::{
+pub(crate) use launch_at_login::{
     ToggleLaunchAtLoginOutcome, disable_launch_at_login, enable_launch_at_login,
     is_launch_at_login_enabled, macos_app_bundle_path, toggle_launch_at_login_outcome,
 };
 
 // Re-exports for update.rs (macOS plist generation).
 #[cfg(target_os = "macos")]
-pub use macos::generate_plist;
+pub(crate) use macos::generate_plist;
 
 #[cfg(target_os = "windows")]
 pub(crate) use windows::kill_freenet_service_processes;
@@ -2882,7 +2882,7 @@ echo "RC=$?"
         let plist_path = home.join("org.freenet.node.plist");
 
         for path in [&wrapper_path, &hash_path, &bak_path, &plist_path] {
-            fs::write(path, b"x").expect("write fixture file");
+            fs::write(path, b"x" as &[u8]).expect("write fixture file");
             assert!(path.exists(), "fixture {} should exist", path.display());
         }
 

--- a/crates/core/src/bin/commands/service/launch_at_login.rs
+++ b/crates/core/src/bin/commands/service/launch_at_login.rs
@@ -5,6 +5,15 @@ use std::path::{Path, PathBuf};
 #[cfg(target_os = "macos")]
 use super::wrapper::log_wrapper_event;
 
+// First-run / legacy-migration marker helpers live in the parent `service`
+// module (platform-independent so they can be unit-tested on Linux CI); the
+// macOS startup self-heal path below references them.
+#[cfg(target_os = "macos")]
+use super::{
+    first_run_marker_path, is_first_run_at, legacy_migration_marker_path,
+    mark_first_run_complete_at,
+};
+
 // ── Launch at Login (macOS) ──
 //
 // On macOS, the DMG-installed Freenet.app does not auto-start on login by
@@ -59,7 +68,7 @@ pub(super) fn is_launch_at_login_enabled_at(plist: &Path) -> bool {
 /// Exposed at `pub(super)` so `update.rs` can detect bundle context for
 /// its DMG-swap auto-update path.
 #[allow(dead_code)]
-pub(super) fn macos_app_bundle_path(exe: &Path) -> Option<PathBuf> {
+pub(crate) fn macos_app_bundle_path(exe: &Path) -> Option<PathBuf> {
     for ancestor in exe.ancestors() {
         if ancestor
             .file_name()
@@ -269,7 +278,7 @@ fn launchctl_bootout(_plist: &Path) {}
 
 /// Enable launch-at-login: write the plist and ask launchd to load it.
 #[allow(dead_code)]
-pub(super) fn enable_launch_at_login(executable: &Path) -> std::io::Result<()> {
+pub(crate) fn enable_launch_at_login(executable: &Path) -> std::io::Result<()> {
     let Some(plist) = launch_agent_plist_path() else {
         return Err(std::io::Error::new(
             std::io::ErrorKind::NotFound,
@@ -289,7 +298,7 @@ pub(super) fn enable_launch_at_login(executable: &Path) -> std::io::Result<()> {
 }
 
 #[allow(dead_code)]
-pub(super) fn disable_launch_at_login() -> std::io::Result<()> {
+pub(crate) fn disable_launch_at_login() -> std::io::Result<()> {
     let Some(plist) = launch_agent_plist_path() else {
         return Ok(());
     };
@@ -299,7 +308,7 @@ pub(super) fn disable_launch_at_login() -> std::io::Result<()> {
 
 /// User-facing query: is launch-at-login currently on?
 #[allow(dead_code)]
-pub(super) fn is_launch_at_login_enabled() -> bool {
+pub(crate) fn is_launch_at_login_enabled() -> bool {
     launch_agent_plist_path()
         .map(|p| is_launch_at_login_enabled_at(&p))
         .unwrap_or(false)
@@ -340,13 +349,13 @@ pub(super) fn first_run_launch_at_login_action(
 /// rationale as `first_run_launch_at_login_action`.
 #[derive(Debug, PartialEq, Eq)]
 #[allow(dead_code)]
-pub(super) enum ToggleLaunchAtLoginOutcome {
+pub(crate) enum ToggleLaunchAtLoginOutcome {
     Enable,
     Disable,
 }
 
 #[allow(dead_code)]
-pub(super) fn toggle_launch_at_login_outcome(
+pub(crate) fn toggle_launch_at_login_outcome(
     currently_enabled: bool,
 ) -> ToggleLaunchAtLoginOutcome {
     if currently_enabled {

--- a/crates/core/src/bin/commands/service/macos.rs
+++ b/crates/core/src/bin/commands/service/macos.rs
@@ -23,7 +23,7 @@ pub(super) fn install_service(system: bool) -> Result<()> {
 /// writes the script), the update path, and the uninstall path (which
 /// removes it). Keep this in sync with `update.rs`'s wrapper derivation.
 #[cfg(target_os = "macos")]
-fn wrapper_script_path(home_dir: &Path) -> PathBuf {
+pub(super) fn wrapper_script_path(home_dir: &Path) -> PathBuf {
     home_dir.join(".local/bin/freenet-service-wrapper.sh")
 }
 
@@ -37,7 +37,7 @@ fn wrapper_script_path(home_dir: &Path) -> PathBuf {
 /// Regression target for #4290: install wrote three files, uninstall
 /// removed zero, leaving stale wrapper artifacts in `~/.local/bin`.
 #[cfg(target_os = "macos")]
-fn remove_wrapper_files(wrapper_path: &Path) -> Result<()> {
+pub(super) fn remove_wrapper_files(wrapper_path: &Path) -> Result<()> {
     use std::fs;
 
     // `.sh` itself, plus the `.sh.hash` sidecar (#4286) and any `.sh.bak`
@@ -352,7 +352,7 @@ done
 }
 
 #[cfg(target_os = "macos")]
-pub(super) fn generate_plist(wrapper_path: &Path, log_dir: &Path) -> String {
+pub(crate) fn generate_plist(wrapper_path: &Path, log_dir: &Path) -> String {
     // Note: wrapper_path is the auto-update wrapper script, not the freenet binary directly.
     // The wrapper handles the loop: run freenet, check exit code, update if needed.
     format!(
@@ -433,7 +433,7 @@ pub fn stop_and_remove_service(_system: bool) -> Result<bool> {
 /// `$HOME` or shelling out to `launchctl` (which only runs when the plist is
 /// present).
 #[cfg(target_os = "macos")]
-fn stop_and_remove_service_at(home_dir: &Path) -> Result<bool> {
+pub(super) fn stop_and_remove_service_at(home_dir: &Path) -> Result<bool> {
     use std::fs;
 
     let plist_path = home_dir.join("Library/LaunchAgents/org.freenet.node.plist");

--- a/crates/core/src/bin/commands/service/wrapper.rs
+++ b/crates/core/src/bin/commands/service/wrapper.rs
@@ -7,11 +7,14 @@ use super::launch_at_login::macos_launch_at_login_startup;
 use super::single_instance::{AcquireWrapperLockOutcome, acquire_wrapper_single_instance_lock};
 
 #[cfg(any(target_os = "windows", target_os = "macos"))]
+use super::single_instance::FIRST_RUN_OPENER_SPAWNED;
+
+#[cfg(any(target_os = "windows", target_os = "macos"))]
 use super::DASHBOARD_URL;
 #[cfg(any(target_os = "windows", target_os = "macos"))]
 use super::open_url_in_browser;
-#[cfg(any(target_os = "windows", target_os = "macos"))]
-use super::single_instance::FIRST_RUN_OPENER_SPAWNED;
+// First-run marker helpers + dashboard reachability probe live in the parent
+// `service` module; the first-run onboarding opener below references them.
 use super::{
     SENTINEL_RESTART, SENTINEL_STOP, WRAPPER_EXIT_ALREADY_RUNNING, WRAPPER_EXIT_UPDATE_NEEDED,
     WRAPPER_INITIAL_BACKOFF_SECS, WRAPPER_MAX_BACKOFF_SECS, WRAPPER_MAX_CONSECUTIVE_FAILURES,

